src/oic/utils/sdb.py

```diff
@@ -1091,3 +964,117 @@ def read(self, token):
             raise WrongTokenType("Not a grant token")
 
         return self._db[key]
+
+    def get_by_sub(self, sub):
```

Comment:
These new methods need annotation and docstrings.

Reply:
OK

Reply:
This is still missing.

Reply:
OK